src/shared/util.js

```diff
@@ -710,19 +710,27 @@ class Util {
   // The SVD can be represented with formula A = USV. We are interested in the
   // matrix S here because it represents the scale values.
   static singularValueDecompose2dScale(m) {
-    const transpose = [m[0], m[2], m[1], m[3]];
+    if (Math.abs(m[1]) <= Number.EPSILON && Math.abs(m[2]) <= Number.EPSILON) {
```

Comment:
The changes in this file probably belongs in a *separate commit*, since they seem orthogonal to the rest of the changes.